docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,265 @@
+|Windows desktops|Users authenticate to the Teleport Web UI. The Web UI initiates a WebSocket session with the Teleport Desktop Service and communicates with the service using Teleport Desktop Protocol (TDP). The Desktop Service converts TDP traffic to Remote Desktop Protocol traffic.|
+
+In most cases (except for some self-hosted clusters), the user will have
+connected to the Auth Service through the Teleport Proxy Service. The Auth
+Service forwards user certificates through the Proxy Service over mutual TLS.  
+
+## Establishing reverse tunnels
+
+In most cases, when an agent joins a Teleport cluster, it establishes a reverse
+SSH tunnel to the Proxy Service. As such, users do not need to have direct
+connectivity to agents or the resources they are connected to. As long as an
+agent can dial back to the cluster's Proxy Service, it can be located behind a
+firewall.
+
+![Tunnel Mode](../../img/architecture/k8s-tunnel.png)
+
+The connection between the Proxy Service and a Teleport agent instance is also
+authenticated with mutual TLS. The Proxy Service generates a short-lived X.509
+certificate signed by the cluster's host CA, with the client's identity and
+routing information for an infrastructure resource encoded in it, and uses it to
+authenticate with the Teleport agent.
+
+The Proxy Service authenticates the connection and dispatches it to the
+appropriate agent instance based on the routing information encoded in the
+client certificate, over the reverse tunnel.
+
+<Details title="Connecting to agents without reverse tunnels">
+
+It is possible to join Teleport agents to a cluster [through the Teleport Auth
+Service](../agents/join-services-to-your-cluster/join-token.mdx#connecting-directly-to-the-auth-service.mdx).
+Once an agent joins a cluster through the Teleport Auth Service, the Teleport
+Proxy Service dials the agent directly, without creating a reverse tunnel. This
+mode supports the following services:
+
+- Teleport SSH Service
+- Teleport Desktop Service
+- Teleport Kubernetes Service
+- Teleport Discovery Service
+
+In direct mode, SSH Service instances act like OpenSSH servers that only accept
+client SSH certificates. Users can connect to SSH servers through the Teleport
+Proxy Service as a jump-host or directly:
+
+![Standard Mode](../../img/architecture/ssh-direct-mode@1.2x.svg)
+
+Direct mode is designed for legacy use cases and only supports self-hosted
+Teleport clusters. We recommend joining agents through the Teleport Proxy
+Service and the reverse tunnel system unless this is not possible for your
+self-hosted Teleport deployment.
+
+</Details>
+
+## Connecting to infrastructure through Teleport agents
+
+Teleport users can access resources in their infrastructure with the `tsh`
+clientCLI, the Teleport Web UI, and Teleport Connect. All of these tools route
```

Comment:
Can't they also access resources with standard clients for some features, e.g. `psql`, `kubectl`?